File: ReleaseNotes.API/ReleaseNotes.API/Services/Bundle/Models/BundleResponseModel.cs

using ReleaseNotes.API.Data.Models;
using ReleaseNotes.API.Services.Project.Models.ReleaseNotes;

namespace ReleaseNotes.API.Services.Bundle.Models;

public class BundleResponseModel
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTimeOffset CreatedOnUtc { get; set; }
    public string CreatedBy { get; set; } = string.Empty;
    public List<ReleaseNotesResponseModel> Projects { get; set; } = [];
    public List<BundleReleaseTimeRangesResponseModel> Releases { get; set; } = [];
}

File: ReleaseNotes.UI/src/services/bundle/models/bundle-response-model.ts
import { ReleaseNotesResponseModel } from "../../project/models/release-notes/release-notes-response-model";
import { BundleReleaseTimeRangesResponseModel } from "./bundle-release-time-ranges-response-model";

export interface BundleResponseModel {
  id: string;
  name: string;
  CreatedOnUtc: Date;
  createdBy: string;
  projects: ReleaseNotesResponseModel[];
  releases: BundleReleaseTimeRangesResponseModel[];
}
